compiler/rustc_resolve/src/macros.rs

```diff
@@ -500,6 +505,28 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             Err(Determinacy::Undetermined) => return Err(Indeterminate),
         };
 
+        if self.tcx.features().diagnostic_namespace
+            && kind == MacroKind::Attr
+            && !path.segments.is_empty()
+            && path.segments[0].ident.as_str() == "diagnostic"
+        {
+            if path.segments.len() != 2 {
+                self.tcx
+                    .sess
+                    .span_err(path.span, "Diagnostic attributes requires a path with 2 segments");
```

Comment:
We don't use title case in errors, so this would start as "diagnostic attributes". As this is referring to a name, we could go as far as using backticks around it.